File: anserial.go
// anserial project main.go
package main

import (
	"fmt"
	"log"
	"os"
	"strings"
	"unicode/utf8"
)

const digits = "0123456789ABCDEF"

//const digits = "ABCDEF"
const prefix = ""

type Anserial struct {
	digits string
	prefix string
}

func (ans Anserial) getNext(anum string) string {
	dlen := utf8.RuneCountInString(digits)
	plen := utf8.RuneCountInString(ans.prefix)
	bnum := string([]rune(anum)[plen:])
	blen := utf8.RuneCountInString(bnum)
	//bnum := anum

	validnum := true
	for _, rv := range bnum {
		if strings.IndexRune(digits, rv) < 0 {
			validnum = false
			break
		}
	}
	if !validnum {
		return ans.prefix + string(digits[0])
	}

	zero := false
	if string(digits[0]) == "0" {
		zero = true
	}
	overflow := false
	for i, _ := range bnum {
		pos := blen - 1 - 1*i
		adigit := string([]rune(bnum)[pos])
		idx := strings.Index(digits, adigit)
		if idx < 0 {
			return ans.prefix + string(digits[0])
		}
		if idx == dlen-1 {
			overflow = true
		}
		//		fmt.Println(string(digits[0]))
		if pos == blen-1 {
			if idx == dlen-1 {
				bnum = string([]rune(bnum)[:pos]) + string([]rune(digits)[0])
				overflow = true
				continue
			} else {
				bnum = bnum[:pos] + string([]rune(digits)[idx+1])
				return ans.prefix + bnum

			}
		}
		if overflow {
			if idx == dlen-1 {
				bnum = string([]rune(bnum)[:pos]) + string([]rune(digits)[0]) + string([]rune(bnum)[pos+1:])
				overflow = true
				continue
			} else {
				bnum = string([]rune(bnum)[:pos]) + string([]rune(digits)[idx+1]) + string([]rune(bnum)[pos+1:])
				return ans.prefix + bnum
			}
		} else {
			return ans.prefix + bnum

		}

	}

	if overflow {
		if zero {
			bnum = string([]rune(digits)[1]) + bnum
		} else {
			bnum = string([]rune(digits)[0]) + bnum
		}
	}

	return ans.prefix + bnum
}

func main() {
	ans := Anserial{digits, prefix}
	num := prefix + string(digits[0])
	f, err := os.Create("anserial.txt")
	if err != nil {
		log.Println(err)
		f.Close()
		return
	}
	for i := 1; i < 10000; i++ {
		fmt.Println(i, num)
		_, err = f.WriteString(fmt.Sprintf("%d %s\n", i, num))
		if err != nil {
			log.Println(err)
			f.Close()
			return
		}
		num = ans.getNext(num)
	}
	err = f.Close()
	if err != nil {
		log.Println(err)
		return
	}
	fmt.Println("Done")
}

File: anserial.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
#----------------------------------------------------------------------------
# Name:         Anserial
# Author:       Doseok Kim
# Copyright:    (c)Madang Business Solutions LLC
#----------------------------------------------------------------------------

class AlphaNum():
  def __init__(self):
    self.digits = "0123456789ABCDEF"
    self.dlen = len(self.digits)
    
  def getNext(self,anumber):      
    bnumber = anumber.upper()
    if not bnumber:
      bnumber = self.digits[0]
    blen = len(bnumber)
    overflow = False
    for i in range(blen):
      pos = -1-1*i
      adigit = bnumber[pos]
      try:
        idx = self.digits.index(adigit)
      except:
        return "0"
      if idx==len(self.digits)-1:
        overflow = True
      #print(blen+pos==blen-1)
      #print(pos,idx,overflow,bnumber)
      
      if blen+pos==blen-1:
        if idx==self.dlen-1:
          bnumber=bnumber[:pos]+self.digits[0]
          overflow = True
          continue
        else:
          bnumber=bnumber[:pos]+self.digits[idx+1]
          return bnumber
      if overflow:
        #print(bnumber,idx,pos,bnumber[:pos],self.digits[0],bnumber[pos+1:],)
        if idx==self.dlen-1:
          bnumber=bnumber[:pos]+self.digits[0]+bnumber[pos+1:]
          overflow = True
          continue
        else:
          bnumber=bnumber[:pos]+self.digits[idx+1]+bnumber[pos+1:]
          return bnumber
      else:
        return bnumber
    if overflow:
      bnumber = self.digits[1]+bnumber
    return bnumber
    
if __name__=="__main__":
  an = Anserial()
  anum = ""
  with open("output.txt","wb") as f:
    for i in range(10000):
      #print(anum)
      anum = an.getNext(anum)
      f.write(bytes("%s\n"%(anum,),"utf8"))
